holopy/core/tests/test_metadata.py

```diff
@@ -0,0 +1,666 @@
+        normals /= np.linalg.norm(normals, axis=0).reshape(1, -1)
+        test = update_metadata(points, normals=normals)
+        self.assertTrue(np.all(test.normals.values == normals))
+
+    @attr("fast")
+    def test_raises_error_when_normals_are_incorrect_shape_for_grid(self):
+        detector = detector_grid(3, 0.1)
+        np.random.seed(13)
+        normals = np.random.randn(4)
+        normals /= np.linalg.norm(normals)
+        self.assertRaises(
+            ValueError, update_metadata, detector, normals=normals)
+
+    @attr("fast")
+    def test_raises_error_when_normals_are_transposed_for_points(self):
+        npts = 12
+        x = np.random.randn(npts)
+        y = np.random.randn(npts)
+        z = np.random.randn(npts)
+        points = detector_points(x=x, y=y, z=z)
+
+        np.random.seed(13)
+        normals = np.random.randn(npts, 3)
+        normals /= np.linalg.norm(normals, axis=1).reshape(-1, 1)
+        self.assertRaises(
+            ValueError, update_metadata, points, normals=normals)
+
+    @attr("fast")
+    def test_raises_error_when_normals_are_incorrect_shape_for_points(self):
+        npts = 12
+        x = np.random.randn(npts)
+        y = np.random.randn(npts)
+        z = np.random.randn(npts)
+        points = detector_points(x=x, y=y, z=z)
+
+        np.random.seed(13)
+        normals = np.random.randn(3, npts + 1)
+        normals /= np.linalg.norm(normals, axis=0).reshape(1, -1)
+        self.assertRaises(
+            ValueError, update_metadata, points, normals=normals)
+
+
+class TestGetSpacing(unittest.TestCase):
+    @attr("fast")
+    def test_raises_error_when_xspacing_is_unequal(self):
+        x = np.linspace(0, 1, 11)**2  # non-uniform spacing
+        y = np.linspace(0, 1, 11)
+
+        detector = detector_points(x=x, y=y)
+        self.assertRaises(ValueError, get_spacing, detector)
+
+    @attr("fast")
+    def test_raises_error_when_yspacing_is_unequal(self):
+        x = np.linspace(0, 1, 11)
+        y = np.linspace(0, 1, 11)**2  # non-uniform spacing
+
+        detector = detector_points(x=x, y=y)
+        self.assertRaises(ValueError, get_spacing, detector)
+
+    @attr("fast")
+    def test_on_different_spacings(self):
+        xspacing = 0.1
+        yspacing = 0.2
+        detector = detector_grid((10, 10), spacing=(xspacing, yspacing))
+
+        spacing = get_spacing(detector)
+        self.assertEqual(spacing[0], xspacing)
+        self.assertEqual(spacing[1], yspacing)
+
+    @attr("fast")
+    def test_on_same_spacing(self):
+        true_spacing = 0.1
+        detector = detector_grid((10, 10), spacing=true_spacing)
+
+        spacing = get_spacing(detector)
+        self.assertEqual(spacing[0], true_spacing)
+        self.assertEqual(spacing[1], true_spacing)
+
+
+class TestGetExtents(unittest.TestCase):
+    @attr('fast')
+    def test_returns_empty_when_dims_is_point(self):
+        # FIXME is this the desired behavior, to return {}?
```

Comment:
5. Is it the desired behavior for ```get_extents``` to return an empty dict when the detector is from ```detector_points```?

Reply:
Should probably raise an error if not a detector_grid.

Reply:
Should be fixed by 1faa266